### PR TITLE
Add support to 2025 compiler

### PR DESCRIPTION
**Pull request by ZzEeKkAa**

### CODE REVIEW

**Comment by tdeng5** [comment]
Could we use __LIBSYCL_MAJOR_VERSION to identify the oneAPI version?
 
reference PR: [adapt compiler version (#39) · intel/xetla@ae46a69](https://github.com/intel/xetla/commit/ae46a690bac364a93437e248418636c2a8423134)
adapt compiler version (#39) · intel/xetla@ae46a69

File: include/common/core/common.hpp [hunk 1]
```diff
@@ -224,8 +224,11 @@ enum class reduce_op : uint8_t {
 
 /// SW_BARRIER, insert software scheduling barrier, for better code control
 ///
-
-#define SW_BARRIER() __ESIMD_NS::fence<__ESIMD_NS::fence_mask::sw_barrier>()
+#ifdef __SYCL_DEVICE_ONLY__
```
Comment (anmyachev) [inline include/common/core/common.hpp:227]: ~@ZzEeKkAa could you provide more details about this change?~

File: include/subgroup/tile/impl/payload_xe.hpp [hunk 2]
```diff
@@ -79,7 +79,7 @@ struct mem_payload_t<
     xetla_vector<uint32_t, 16 * num_block> payloads;
 
     inline mem_payload_t(const this_payload_t &rhs) {
-        this->payload = rhs.payload;
+        this->payloads = rhs.payloads;
```
Comment (anmyachev) [inline include/subgroup/tile/impl/payload_xe.hpp:82]: It was definitely a bug.

**Comment by tdeng5** [comment]
test xetla pr

**Review by tdeng5 — APPROVED** [review]

**Pull request merged**